### PR TITLE
init redistimeseries at 1.10.10 and add redisModules

### PR DESCRIPTION
**Pull request by de11n**
## Description of changes

Adds [redistimeseries](https://redis.io/docs/data-types/timeseries/), the RSALv2 license that it uses, and a new redisModules top-level attribute to collect this and other redis modules. I could imagine a `redis.withModules` function at some point.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: lib/licenses.nix [hunk 1]
```diff
@@ -1019,6 +1019,11 @@ in mkLicense lset) ({
     url = "https://qwt.sourceforge.io/qwtlicense.html";
   };
 
+  rsal2 = {
+    fullName = "Redis Source Available License 2.0";
+    url = "https://redis.com/legal/rsalv2-agreement";
```
Comment (alyssais) [inline lib/licenses.nix:1024]:
```suggestion
    url = "https://redis.com/legal/rsalv2-agreement";
    free = false;
```

**Review by alyssais — CHANGES_REQUESTED** [review]

File: lib/licenses.nix (hunk 1 above)
Comment (de11n) [reply lib/licenses.nix:1024]: How are you arriving at this suggestion? It was not obvious to me that this is "unfree." Anyone can use this software without paying. They merely cannot commercialize it directly.

Comment (alyssais) [reply lib/licenses.nix:1024]: "Free" here does not mean "gratis".  It's used in the sense of [Free Software](https://en.wikipedia.org/wiki/Free_software), which has a quite specific meaning to regarding what freedoms need to be provided to users for the term to apply, and one of those is the freedom to use the program for any purpose, including direct commercialization.

If you look through the rest of the licenses marked as not free in the list, you'll see that very few of them are about software that has a licensing fee attached, but that they all have conditions that restrict users' freedom to the point that they don't count as free software.

Comment (de11n) [reply lib/licenses.nix:1029]: Thank you! I've updated it.

**Pull request closed without merging**